process/block/sovereignChainBlock.go

```diff
@@ -890,40 +887,33 @@ func (scbp *sovereignChainBlockProcessor) processEpochStartMetaBlock(
 		return err
 	}
 
-	// TODO MX-15589 check how we can integrate all of these later on
-	//err = mp.epochRewardsCreator.VerifyRewardsMiniBlocks(header, allValidatorsInfo, computedEconomics)
-	//if err != nil {
-	//	return err
-	//}
-	//
-	//err = mp.epochSystemSCProcessor.ProcessDelegationRewards(body.MiniBlocks, mp.epochRewardsCreator.GetLocalTxCache())
-	//if err != nil {
-	//	return err
-	//}
+	rewardMiniBlocks, err := scbp.epochRewardsCreator.CreateRewardsMiniBlocks(sovHdr, allValidatorsInfo, &sovHdr.EpochStart.Economics)
+	if err != nil {
+		return err
+	}
+
+	sovHdr.EpochStart.Economics.RewardsForProtocolSustainability.Set(scbp.epochRewardsCreator.GetProtocolSustainabilityRewards())
 
-	//err = mp.validatorInfoCreator.VerifyValidatorInfoMiniBlocks(body.MiniBlocks, allValidatorsInfo)
-	//if err != nil {
-	//	return err
-	//}
+	err = scbp.epochSystemSCProcessor.ProcessDelegationRewards(rewardMiniBlocks, scbp.epochRewardsCreator.GetLocalTxCache())
+	if err != nil {
+		return err
+	}
 
 	err = scbp.epochEconomics.VerifyRewardsPerBlock(sovHdr, scbp.epochRewardsCreator.GetProtocolSustainabilityRewards(), computedEconomics)
 	if err != nil {
 		return err
 	}
-	//
-	//err = mp.verifyFees(header)
-	//if err != nil {
-	//	return err
-	//}
 
-	saveEpochStartEconomicsMetrics(scbp.appStatusHandler, sovHdr)
+	// do NOT call scbp.epochRewardsCreator.VerifyRewardsMiniBlocks here as in original meta code, since that func will re-process rewards
 
-	validatorMiniBlocks, err := scbp.validatorInfoCreator.CreateValidatorInfoMiniBlocks(allValidatorsInfo)
+	err = scbp.verifyFees(header)
```

Comment:
I don't think you need this here. This verification is done also in `scbp.verifySovereignPostProcessBlock`

Reply:
We need this for epoch start, since we do not call that function on epoch start processing:

``` go
	if shardHeader.IsStartOfEpochBlock() {
		err = scbp.processEpochStartMetaBlock(shardHeader, body)
		if err != nil {
			return nil, nil, err
		}

		body, err = scbp.applyBodyToHeader(shardHeader, body)
		if err != nil {
			return nil, nil, err
		}

		return shardHeader, body, nil
	}
```

Reply:
Right